internal/parser/parser.go

```diff
@@ -21,6 +26,809 @@ type Scanner interface {
+	switch pos, tok, lit := p.scanWithRegex(); tok {
+	case token.IDENT:
+		return &ast.Identifier{Name: lit}
+	case token.INT:
+		value, err := strconv.ParseInt(lit, 10, 64)
+		if err != nil {
+			panic(err)
+		}
+		return &ast.IntegerLiteral{Value: value}
+	case token.FLOAT:
+		value, err := strconv.ParseFloat(lit, 64)
+		if err != nil {
+			panic(err)
+		}
+		return &ast.FloatLiteral{Value: value}
+	case token.STRING:
+		value, err := strconv.Unquote(lit)
+		if err != nil {
+			panic(err)
+		}
+		return &ast.StringLiteral{Value: value}
+	case token.REGEX:
+		value, err := parseRegexp(lit)
+		if err != nil {
+			panic(err)
+		}
+		return &ast.RegexpLiteral{Value: value}
+	case token.DURATION:
+		values, err := parseDuration(lit)
+		if err != nil {
+			panic(err)
+		}
+		return &ast.DurationLiteral{Values: values}
+	case token.PIPE_RECEIVE:
+		return &ast.PipeLiteral{}
+	case token.LBRACK:
+		p.unread(pos, tok, lit)
+		return p.arrayLiteral()
+	case token.LBRACE:
+		p.unread(pos, tok, lit)
+		return p.objectLiteral()
+	case token.LPAREN:
+		p.unread(pos, tok, lit)
+		return p.parenExpression()
+	default:
+		return nil
+	}
+}
+
+func (p *parser) identifier() *ast.Identifier {
+	_, lit := p.expect(token.IDENT)
+	return &ast.Identifier{Name: lit}
+}
+
+func (p *parser) arrayLiteral() ast.Expression {
+	p.expect(token.LBRACK)
+	exprs := p.expressionList()
+	p.expect(token.RBRACK)
+	return &ast.ArrayExpression{Elements: exprs}
+}
+
+func (p *parser) objectLiteral() ast.Expression {
+	p.expect(token.LBRACE)
+	properties := p.propertyList()
+	p.expect(token.RBRACE)
+	return &ast.ObjectExpression{Properties: properties}
+}
+
+func (p *parser) parenExpression() ast.Expression {
+	p.expect(token.LPAREN)
+	return p.parenExpressionBody()
+}
+
+func (p *parser) parenExpressionBody() ast.Expression {
+	switch _, tok, lit := p.peek(); tok {
+	case token.RPAREN:
+		p.consume()
+		return p.arrowExpression(nil)
+	case token.IDENT:
+		p.consume()
+		ident := &ast.Identifier{Name: lit}
+		return p.parenIdentExpression(ident)
+	default:
+		expr := p.expression()
+		p.expect(token.RPAREN)
+		return expr
+	}
+}
+
+func (p *parser) parenIdentExpression(key *ast.Identifier) ast.Expression {
+	switch _, tok, _ := p.peek(); tok {
+	case token.RPAREN:
+		p.consume()
+		if _, tok, _ := p.peek(); tok == token.ARROW {
+			return p.arrowExpression([]*ast.Property{
+				{Key: key},
+			})
+		}
+		return key
+	case token.ASSIGN:
+		p.consume()
+		value := p.expression()
+		params := []*ast.Property{{Key: key, Value: value}}
+		if _, tok, _ := p.peek(); tok == token.COMMA {
+			p.consume()
+			params = append(params, p.parameterList()...)
+		}
+		p.expect(token.RPAREN)
+		return p.arrowExpression(params)
+	case token.COMMA:
+		p.consume()
+		params := []*ast.Property{{Key: key}}
+		params = append(params, p.parameterList()...)
+		p.expect(token.RPAREN)
+		return p.arrowExpression(params)
+	default:
+		expr := p.expressionSuffix(key)
```

Comment:
The ExpressionSuffix is not an alternate in the ParenIdentExpression rule, does the grammar need to be updated?

Reply:
Fixed.